Issue: #503 Allow to serve API without WebUI files

jtorvald opened the pull request:
Added two flags/config options for host_ui and host_docs default to true for backwards compatibility. And fixed link to documentation overview from the UI (pointed to readme, which gives an index of the docs).

dennwc committed:
cayley: switch to cobra/viper cli

cayleyupgrade is now included into main binary

add command to convert quad files without loading them

config format changed, but old format is still supported

updated docs

Commit:
Overview did not exist

Commit:
Issue: #503 Allow to serve API without WebUI files
- added two flags/config options for host_ui and host_docs default to true for backwards compatibility

Commit:
Moved http consts to http and fixed todo about loading database while configuration says: readonly

dennwc commented on internal/http/http.go:
```diff
@@ -152,9 +152,21 @@ func SetupRoutes(handle *graph.Handle, cfg *config.Config) {
-	http.Handle("/", r)
+
+	// only show docs when allowed
+	if cfg.HostDocs {
```
It's might be better to make it a string flag (path to docs directory). And if it's empty docs will not be server.

jtorvald replied:
For the path in the API or the path on disk?
And what about the flag for the UI?

dennwc commented:
On disk. And the same for UI files, sure.

пт, 17 февр. 2017 г. в 0:41, Jørgen <notifications@github.com>:

> *@jtorvald* commented on this pull request.
> ------------------------------
>
> In internal/http/http.go
> <https://github.com/dennwc/cayley/pull/1#discussion_r101645832>:
>
> > @@ -152,9 +152,21 @@ func SetupRoutes(handle *graph.Handle, cfg *config.Config) {
>
>  	//m.Use(martini.Static("static", martini.StaticOptions{Prefix: "/static", SkipLogging: true}))
>  	//r.Handler("GET", "/static", http.StripPrefix("/static", http.FileServer(http.Dir("static/"))))
> -	r.GET("/docs/:docpage", docs.ServeHTTP)
> -	r.GET("/ui/:ui_type", root.ServeHTTP)
> -	r.GET("/", root.ServeHTTP)
> -	http.Handle("/static/", http.StripPrefix("/static", http.FileServer(http.Dir(fmt.Sprint(assets, "/static/")))))
> -	http.Handle("/", r)
> +
> +	// only show docs when allowed
> +	if cfg.HostDocs {
>
> For the path in the API or the path on disk?
> And what about the flag for the UI?
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/dennwc/cayley/pull/1#discussion_r101645832>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AApTdMUZT0wh-ipfvgGnBhtWAa_waR-mks5rdNCngaJpZM4MDPNv>
> .
>

Commit:
Fixed the listening to host/port from config or command with default to 127.0.0.1:64210
Also prepared hosting docs and ui in different path, but need to finish it later.
Updated configuration for listen and hosting ui/docs (preparation)

dennwc commented on cmd/cayley/cayley.go:
```diff
@@ -115,15 +115,21 @@ func init() {
-	rootCmd.PersistentFlags().Bool("host_docs", true, "host documentation")
+	// http
+	rootCmd.PersistentFlags().String("listen", "", "host to listen on")
+	rootCmd.PersistentFlags().String("path_ui", "/ui/", "host web ui")
```
should be `./ui/` and `./docs`

dennwc commented on cmd/cayley/command/http.go:
```diff
@@ -59,23 +63,43 @@ func NewHttpCmd() *cobra.Command {
 		},
 	}
-	cmd.Flags().String("host", ":64210", "host:port to listen on")
+	cmd.Flags().String("listen", "", "host:port to listen on")
```
please leave it as `host`, I think it's more common to name it like this.

dennwc commented on cmd/cayley/command/http.go:
```diff
@@ -59,23 +63,43 @@ func NewHttpCmd() *cobra.Command {
+
+			// do we have a port, otherwise add default port
+			var findPort = regexp.MustCompile(`:[0-9]+$`)
+			if !findPort.MatchString(listen) {
```
SplitHostPort should work fine here

dennwc commented on internal/http/http.go:
```diff
@@ -145,27 +145,27 @@ func SetupRoutes(handle *graph.Handle, cfg *config.Config) {
-	root := &TemplateRequestHandler{templates: templates}
-	docs := &DocRequestHandler{assets: assets}
+	root := &TemplateRequestHandler{templates: templates, path: cfg.HostUI}
+	docs := &DocRequestHandler{assets: assets, path: cfg.HostDocs}
```
Assets is the same parameter, no?

Commit:
fixed assets folder configureable and as cli argument + documentation + documentation should not display in UI if serve_docs is turned off. This needs work.

Commit:
Added myself as a contributor to the project
Updated example config
Fixed the docs (links with .md like in the README did not work)
Not showing documentations button if docs are not served
Added some extra English text to write.html because Chrome kept offering to translate from Portuguese to English
